test/features/users.feature

```diff
@@ -0,0 +1,91 @@
+        Given "Group Admin" as the persona
+        When I log in
+        And I go to the user list API
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": true,') and contains(string(), '"name": "admin"')]"
+
+    Scenario: Ensure user_list is not accessible to non-admins
+        Given "Publisher" as the persona
+        When I log in
+        And I go to the user list API
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": false,') and contains(string(), 'Authorization Error')]"
+
+    Scenario: Ensure user_list is not accessible anonymously
+        When I go to the user list API
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": false,') and contains(string(), 'requires an authenticated user')]"
+
+
+    Scenario: Ensure organisation membership is accessible to sysadmins
+        Given "Admin" as the persona
+        When I log in
+        And I view the "department-of-health" organisation API including users
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": true,') and contains(string(), '"name": "group_admin"') and contains(string(), '"name": "publisher"')]"
+
+    Scenario: Ensure organisation membership is accessible to organisation admins
+        Given "Group Admin" as the persona
+        When I log in
+        And I view the "department-of-health" organisation API including users
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": true,') and contains(string(), '"name": "group_admin"') and contains(string(), '"name": "publisher"')]"
+
+    Scenario: Ensure organisation membership is not accessible to non-admin members
+        Given "Publisher" as the persona
+        When I log in
+        And I view the "department-of-health" organisation API including users
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": false,') and contains(string(), 'Authorization Error')]"
+
+    Scenario: Ensure organisation membership is not accessible to admins of other organisations
+        Given "Foodie" as the persona
+        When I log in
+        And I view the "department-of-health" organisation API including users
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": false,') and contains(string(), 'Authorization Error')]"
+
+    Scenario: Ensure organisation membership is not accessible anonymously
+        When I view the "department-of-health" organisation API including users
+        And I take a screenshot
+        Then I should see an element with xpath "//*[contains(string(), '"success": false,') and contains(string(), 'Authorization Error')]"
```

Comment:
add positive tests for viewing organisation api or group api with user viewing false for all personas that it works (including anon)

Reply:
Huh, that actually caught a legitimate bug. It seems that custom auth functions need to be annotated to permit anonymous access.